library/pool.c

```diff
@@ -55,7 +55,10 @@ static void dnet_work_pool_stop(struct dnet_work_pool_place *place)
 
 	for (i = 0; i < place->pool->num; ++i) {
 		wio = &place->pool->wio_list[i];
-		pthread_join(wio->tid, NULL);
+		if (!wio->joined) {
+			pthread_join(wio->tid, NULL);
+			wio->joined = 1;
+		}
```

Comment:
>since pool can grab io requests from the queue and start processing them, if you've removed cache, it will crash.

Pool will not grab any request, because it was stopped with dnet_node_stop_common_resources()